### ISSUE TITLE
Important Exception Being Swallowed

### ISSUE THREAD

**Issue by DavePearce** [Whiley/WhileyCompiler#819]
_(Thanks to @utting for pointing this out)_

Currently, the Whiley compiler is swallowing some potentially important exceptions which is a major problem.  The following illustrates.  First we prove we can compile a file:

```
bash-3.2$ wy compile --verbose test.whiley 

WARNING: version numbering unavailable
Parsed 1 source file(s). ....................................................... [41ms+1mb]
Typed 1 source file(s). ............................................................ [13ms]
Generated code for 1 source file(s). ............................................... [11ms]
Whiley => Wyil: compiled 1 file(s) ............................................. [67ms+3mb]
```
Then, we delete the standard library:
```
bash-3.2$ rm ~/projects/WhileyDevelopmentKit/dist/wdk-v0.5.0/lib/wystd-v0.2.2.jar 
```

Finally, we try to compile our file again (after deleting the `wyil` file generated before)
```
bash-3.2$ wy compile --verbose test.whiley 

WARNING: version numbering unavailable
bash-3.2$ ls test.wyil
ls: test.wyil: No such file or directory
```
The frustrating thing here is that it _doesn't report any problem whatsoever_.

Specifically, an important exception is being swallowed in `wyc.command.Compile`:

```
} catch (Exception e) {
   // FIXME: this is a problem because it is swallowing exceptions!!
   return Result.INTERNAL_FAILURE;
}
```

The obvious fix is just to insert `e.printStackTrace()` for now.

**Comment by DavePearce** [comment]
Right, error handling now supported with this:

```
// FIXME: need a better error reporting mechanism
System.err.println("internal failure: " + e.getMessage());
if (verbose) {
   printStackTrace(syserr, e);
}
return Result.INTERNAL_FAILURE;
```

At the comment implies, we will eventually need a better mechanism here.

**Comment by DavePearce** [comment]
Ok, this fix is now live as `wyc-0.5.3`
